packages/cli/src/commands/deploy.ts

```diff
@@ -33,19 +35,99 @@ export const generateBundleZip = async (rootDir: string) => {
   zip.addLocalFolder(distDir, 'output');
   zip.writeZip(outputZip);
 
-  consola.success(`Created ${outputZip}`);
+  consola.success(`Bundle created at: ${outputZip}`);
+};
+
+export const generateUploadSignature = async (
+  storeHash: string,
+  accessToken: string,
+  apiHost: string,
+) => {
+  consola.info('Generating upload signature...');
+
+  try {
+    const response = await fetch(
+      `https://${apiHost}/stores/${storeHash}/v3/headless/deployments/uploads`,
+      {
+        method: 'POST',
+        headers: {
+          'X-Auth-Token': accessToken,
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+        body: JSON.stringify({}),
+      },
+    );
+
+    if (!response.ok) {
+      consola.error(`Failed to fetch upload signature: ${response.status} ${response.statusText}`);
+      process.exit(1);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const { data } = (await response.json()) as BigCommerceResponse<{
+      upload_url: string;
+      upload_uuid: string;
+    }>;
```

Comment:
🍹Should we take the opportunity to install `zod`?

Reply:
I thought `zod` was used to validate input? 🤔

Reply:
`zod` should be used at boundaries where you don't know the returned type, so this is a perfect place to use it.

Reply:
🤦 😆

Reply:
Following this pattern @chanceaclark suggested to be able to interact with the `res` if needed.

```
    const res: unknown = await response.json();
    const { data } = UploadSignatureSchema.parse(res);
```